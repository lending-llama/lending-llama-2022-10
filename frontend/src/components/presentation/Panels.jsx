import React from "react";

/*
 cp. https://tailwindui.com/components/application-ui/layout/panels
 */

export const Card = ({children}) =>
  <div className="bg-white overflow-hidden shadow rounded-lg">
    <div className="px-4 py-5 sm:p-6">
      {children}
    </div>
  </div>

export const CardWithHeader = ({header, children}) =>
  <div className="bg-white overflow-hidden shadow rounded-lg divide-y divide-gray-200">
    <div className="px-4 py-5 sm:px-6">
      {header}
    </div>
    <div className="px-4 py-5 sm:p-6">
      {children}
    </div>
  </div>
