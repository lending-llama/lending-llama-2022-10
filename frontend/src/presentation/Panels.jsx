import React from "react";

/*
 cp. https://tailwindui.com/components/application-ui/layout/panels
 */

export const Card = ({header, children}) =>
  <div className={ `bg-white overflow-hidden shadow rounded-lg${header ? " divide-y divide-gray-200" : ""}`}>
    {header ? <div className="px-4 py-5 sm:px-6">
      {header}
    </div> : null}
    <div className="px-4 py-5 sm:p-6">
      {children}
    </div>
  </div>
