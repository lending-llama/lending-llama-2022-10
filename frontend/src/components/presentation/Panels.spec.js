import React from 'react';
import renderer from 'react-test-renderer';
import {Card, CardWithHeader} from "./Panels";

describe('Card', () => {
  it('has skeleton as specified by Tailwind UI', () => {
    expect(renderer.create(<Card/>).toJSON()).toMatchSnapshot();
  })
})

describe('CardWithHeader', () => {
  it('has skeleton as specified by Tailwind UI', () => {
    expect(renderer.create(
      <CardWithHeader header="header">
        main content
      </CardWithHeader>
    ).toJSON()).toMatchSnapshot();
  })
})
