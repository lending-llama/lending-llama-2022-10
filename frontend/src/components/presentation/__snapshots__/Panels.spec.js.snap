// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card has skeleton as specified by Tailwind UI 1`] = `
<div
  className="bg-white overflow-hidden shadow rounded-lg"
>
  <div
    className="px-4 py-5 sm:p-6"
  />
</div>
`;

exports[`CardWithHeader has skeleton as specified by Tailwind UI 1`] = `
<div
  className="bg-white overflow-hidden shadow rounded-lg divide-y divide-gray-200"
>
  <div
    className="px-4 py-5 sm:px-6"
  >
    header
  </div>
  <div
    className="px-4 py-5 sm:p-6"
  >
    main content
  </div>
</div>
`;
